sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -95,15 +98,27 @@ internal sealed class ServiceBusListener : IListener, IScaleMonitorProvider
                     return messagingProvider.CreateSessionMessageProcessor(_client.Value,_entityPath, sessionProcessorOptions);
                 });
 
+            _serviceBusMetricsReceiver = new ServiceBusMetricsReceiver(connection, _entityPath, entityType, _batchReceiver, clientFactory, loggerFactory);
+
             _scaleMonitor = new Lazy<ServiceBusScaleMonitor>(
                 () => new ServiceBusScaleMonitor(
                     functionId,
-                    entityType,
                     _entityPath,
-                    connection,
-                    _batchReceiver,
-                    loggerFactory,
-                    clientFactory));
+                    _serviceBusMetricsReceiver,
+                    loggerFactory
+                    ));
+
+            if (singleDispatch)
```

Comment:
Why do we only support this for single dispatch? For batch, wouldn't concurrency be MaxMessageBatchSize?

Reply:
MaxMessageBatchSize limits the batch size, actual batch size can by anything between [1..MaxMessageBatchSize].
A function app can have the triggers with or without TBS so SC will handle this case.